### PR TITLE
Implement settings api on application route for social links

### LINKED ISSUE #855: Implement settings api on application route to get social links for footer
**I'm submitting a ...**  (check one with "x")
- [ ] bug report
- [x] feature request
- [ ] support request => Please do not submit support requests here, instead ask your query in out Gitter channel at https://gitter.im/fossasia/open-event-frontend

Settings defined in admin/content are used for the footer links.

### PR DESCRIPTION
**Pull request by CosmicCoder96**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:

Implements settings api on application route for social links

Fixes #855

### CODE REVIEW

**Comment by niranjan94** [comment]
Deployment link: https://pr-856-open-event-frontend-fossasia.surge.sh

File: app/templates/components/footer-main.hbs [hunk 1]
```diff
@@ -43,10 +43,25 @@
 
     <div class="three wide column">
       <div class="ui inverted link list">
-        <strong class="item">{{t 'Follow us on'}}</strong>
-        <a class="item" href="#" target="_blank" rel="noopener noreferrer"><i class="facebook f icon"></i> Facebook</a>
-        <a class="item" href="#" target="_blank" rel="noopener noreferrer"><i class="twitter icon"></i> Twitter</a>
-        <a class="item" href="#" target="_blank" rel="noopener noreferrer"><i class="github icon"></i> Github</a>
+        <strong class="item">{{t 'Connect with us'}}</strong>
+        <a class="item" href="#" target="{{socialLinks.supportUrl}}" rel="noopener noreferrer">
+          <i class="info icon"></i> Support
```
Comment (geekyd) [inline app/templates/components/footer-main.hbs]: Use translation helpers.

File: app/templates/components/footer-main.hbs [hunk 2]
```diff
@@ -43,10 +43,25 @@
 
     <div class="three wide column">
       <div class="ui inverted link list">
-        <strong class="item">{{t 'Follow us on'}}</strong>
-        <a class="item" href="#" target="_blank" rel="noopener noreferrer"><i class="facebook f icon"></i> Facebook</a>
-        <a class="item" href="#" target="_blank" rel="noopener noreferrer"><i class="twitter icon"></i> Twitter</a>
-        <a class="item" href="#" target="_blank" rel="noopener noreferrer"><i class="github icon"></i> Github</a>
+        <strong class="item">{{t 'Connect with us'}}</strong>
+        <a class="item" href="#" target="{{socialLinks.supportUrl}}" rel="noopener noreferrer">
```
Comment (geekyd) [inline app/templates/components/footer-main.hbs]: Check if the url is null or not.

**Comment by CosmicCoder96** [comment]
@geekyd done 👍

**Comment by niranjan94** [comment]
Deployment link: https://pr-856-open-event-frontend-fossasia.surge.sh

File: app/templates/components/footer-main.hbs [hunk 3]
```diff
@@ -43,10 +43,25 @@
...
-        <strong class="item">{{t 'Follow us on'}}</strong>
-        <a class="item" href="#" target="_blank" rel="noopener noreferrer"><i class="facebook f icon"></i> Facebook</a>
-        <a class="item" href="#" target="_blank" rel="noopener noreferrer"><i class="twitter icon"></i> Twitter</a>
-        <a class="item" href="#" target="_blank" rel="noopener noreferrer"><i class="github icon"></i> Github</a>
+        <strong class="item">{{t 'Connect with us'}}</strong>
+        <a class="item" href="{{if socialLinks.supportUrl socialLinks.supportUrl '#'}}" target="_blank" rel="noopener noreferrer">
+          <i class="info icon"></i> {{t 'Support'}}
+        </a>
+        <a class="item" href="{{if socialLinks.facebookUrl socialLinks.facebookUrl '#'}}" target="_blank" rel="noopener noreferrer">
+          <i class="facebook f icon"></i> Facebook
```
Comment (geekyd) [inline app/templates/components/footer-main.hbs]: Use translation helper here.

File: app/templates/components/footer-main.hbs [hunk 4]
```diff
@@ -43,10 +43,25 @@
...
+        </a>
+        <a class="item" href="{{if socialLinks.twitterUrl socialLinks.twitterUrl '#'}}" target="_blank" rel="noopener noreferrer">
+          <i class="twitter icon"></i> Twitter
```
Comment (geekyd) [inline app/templates/components/footer-main.hbs]: Use translation helper here.

File: app/templates/components/footer-main.hbs [hunk 5]
```diff
@@ -43,10 +43,25 @@
...
+        </a>
+        <a class="item" href="{{if socialLinks.githubUrl socialLinks.githubUrl '#'}}" target="_blank" rel="noopener noreferrer">
+          <i class="github icon"></i> Github
```
Comment (geekyd) [inline app/templates/components/footer-main.hbs]: Use translation helper here.

File: app/templates/components/footer-main.hbs [hunk 6]
```diff
@@ -43,10 +43,25 @@
...
+        </a>
+        <a class="item" ref="{{if socialLinks.youtubeUrl socialLinks.youtubeUrl '#'}}" target="_blank" rel="noopener noreferrer">
+          <i class="youtube icon"></i> Youtube
```
Comment (geekyd) [inline app/templates/components/footer-main.hbs]: Use translation helper here.

File: app/templates/components/footer-main.hbs [hunk 7]
```diff
@@ -43,10 +43,25 @@
...
+        </a>
+        <a class="item" href="{{if socialLinks.googleUrl socialLinks.googleUrl '#'}}" target="_blank" rel="noopener noreferrer">
+          <i class="google plus icon"></i> Google +
```
Comment (geekyd) [inline app/templates/components/footer-main.hbs]: Use translation helper here.

File: app/templates/components/footer-main.hbs (hunk 3 above)
Comment (CosmicCoder96) [reply app/templates/components/footer-main.hbs]: it's a translation service not a transliteration one, facebook won't translate to anytthing

File: app/templates/components/footer-main.hbs (hunk 4 above)
Comment (CosmicCoder96) [reply app/templates/components/footer-main.hbs]: @geekyd it's a translation service not a transliteration one, twitter won't translate to anything

File: app/templates/components/footer-main.hbs (hunk 5 above)
Comment (CosmicCoder96) [reply app/templates/components/footer-main.hbs]: same here

File: app/templates/components/footer-main.hbs (hunk 6 above)
Comment (CosmicCoder96) [reply app/templates/components/footer-main.hbs]: same here

File: app/templates/components/footer-main.hbs (hunk 7 above)
Comment (CosmicCoder96) [reply app/templates/components/footer-main.hbs]: same here

**Comment by niranjan94** [comment]
Deployment link: https://pr-856-open-event-frontend-fossasia.surge.sh

File: app/templates/components/footer-main.hbs (hunk 4 above)
Comment (CosmicCoder96) [reply app/templates/components/footer-main.hbs]: wait .. google translation shows.. support of proper nouns .. adding them.

**Comment by niranjan94** [comment]
Deployment link: https://pr-856-open-event-frontend-fossasia.surge.sh

**Comment by CosmicCoder96** [comment]
@geekyd 
> We should not should the social links if the URL is null. What do you think?

Done, please review.

**Comment by niranjan94** [comment]
Deployment link: https://pr-856-open-event-frontend-fossasia.surge.sh

**Comment by CosmicCoder96** [comment]
@geekyd Please review

File: app/routes/application.js [hunk 8]
```diff
@@ -37,7 +37,8 @@ export default Route.extend(ApplicationRouteMixin, {
       notifications,
       pages: this.get('store').query('page', {
         sort: 'index'
-      })
+      }),
+      socialLinks: this.get('store').queryRecord('setting', {})
```
Comment (geekyd) [inline app/routes/application.js]: @CosmicCoder96 Can we use `findAll` here?

Comment (CosmicCoder96) [reply app/routes/application.js]: @geekyd No. A single record is expected.

Comment (geekyd) [reply app/routes/application.js]: Ahhh yeah!

**Pull request merged**